src/main/java/org/opentripplanner/standalone/config/updater_config/UpdaterConfig.java

```diff
@@ -0,0 +1,27 @@
+package org.opentripplanner.standalone.config.updater_config;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.MissingNode;
+import org.opentripplanner.standalone.config.NodeAdapter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UpdaterConfig {
+
+  public static final UpdaterConfig DEFAULT = new UpdaterConfig(
+      MissingNode.getInstance()
+  );
+
+  private final List<UpdaterConfigItem> configItems = new ArrayList<>();
+
+  public UpdaterConfig(JsonNode c) {
+    for (JsonNode updater : c) {
+      configItems.add(new UpdaterConfigItem(new NodeAdapter(updater, null)));
```

Comment:
Here you create the NodeAdapter, but you should never do that - only do that for root config objects. I will try to remember this an prevent someone form doing it later, but this brake the "parent" reference and error reporting mechanish in the NodeAdapter. Also, you have lost track of the source, when you do this.

Reply:
Also, the entire class is unnecessary. This can be put in the NodeAdapter as a general-purpose factory method:
```
    public <T> List<T> asList(String paramName, Function<NodeAdapter, T> factory) {
        List<T> result = new ArrayList<>();
        int i = 0;
        for (JsonNode node : param(paramName)) {
            String pName = paramName + "[" + i + "]";
            NodeAdapter child = new NodeAdapter(node, source, fullPath(pName));
            children.add(child);
            result.add(factory.apply(child));
            ++i;
        }
        return result;
    }
```
Remember to add unit test as well:
```
@Test
    public void objectAsList() {
        NodeAdapter subject  = newNodeAdapterForTest("{ key : [{ a: 'I' }, { a: '2' } ] }");

        List<String> result = subject.asList("key", a -> a.asText("a"));

        assertEquals("[I, 2]", result.toString());
    }
```

Reply:
Then the UpdaterConfigItem can be renamed UpdaterConfig.